core/src/transfer/data.rs

```diff
@@ -39,6 +40,50 @@ pub enum TransactionData {
     /// Additional data added to a transaction, that is not a deployment or a
     /// call.
     Memo(Vec<u8>),
+    /// Data for blob storage together with contract call.    
+    Blob(Vec<BlobData>),
+}
+
+impl TransactionData {
+    /// Return input bytes to hash the `TransactionData`.
+    ///
+    /// Note: The result of this function is *only* meant to be used as an input
+    /// for hashing and *cannot* be used to deserialize the payload again.
+    #[must_use]
+    pub fn signature_message(&self) -> Vec<u8> {
+        let mut bytes = vec![];
+
+        #[allow(clippy::match_same_arms)]
+        match &self {
+            TransactionData::Deploy(d) => {
+                bytes.extend(&d.bytecode.to_hash_input_bytes());
+                bytes.extend(&d.owner);
+                if let Some(init_args) = &d.init_args {
+                    bytes.extend(init_args);
+                }
+            }
+            TransactionData::Call(c) => {
+                bytes.extend(c.contract.as_bytes());
+                bytes.extend(c.fn_name.as_bytes());
+                bytes.extend(&c.fn_args);
+            }
+            TransactionData::Memo(m) => {
+                bytes.extend(m);
+            }
+            TransactionData::Blob(blobs) => {
+                // Let's hash the blobs and use the hash as Memo for tx
```

Comment:
I wouldn't mention Memo here. Let's stick to what's done by the function.
Also, not callers put the result in Memo

Reply:
The idea of the comment was to enforce the idea that the hash is "serialized" like the Memo